spec/linked_list.js

```diff
@@ -0,0 +1,205 @@
+      expect(myLinkedList.list.data)
+        .to.eql('first')
+    })
+  })
+
+  context('insertBefore()', () => {
+    const myLinkedList = new LinkedList()
+    myLinkedList.insert('second')
+
+    it('Inserts a node with value \'first\' before the first node containing ' +
+      '\'second\'', () => {
+      expect(() => myLinkedList.insertBefore('second', 'first'))
+        .to.alter(() => myLinkedList._size(), {from: 1, to: 2})
+      expect(myLinkedList.list.data)
+        .to.eql('first')
+    })
+  })
+
+  context('insertAfter()', () => {
+    const myLinkedList = new LinkedList()
+    myLinkedList.insert('first')
+
+    it('Inserts a node with value \'third\' after the first node containing ' +
+      '\'first\'', () => {
+      expect(() => myLinkedList.insertAfter('first', 'third'))
+        .to.alter(() => myLinkedList._size(), {from: 1, to: 2})
+      expect(myLinkedList.list.data)
+        .to.eql('first')
+      expect(myLinkedList.list.next.data)
+        .to.eql('third')
+    })
+    it('Inserts a node with value \'second\' after the first node containing ' +
+      '\'first\'', () => {
+      expect(() => myLinkedList.insertAfter('first', 'second'))
+        .to.alter(() => myLinkedList._size(), {from: 2, to: 3})
+      expect(myLinkedList.list.data)
+        .to.eql('first')
+      expect(myLinkedList.list.next.data)
+        .to.eql('second')
+      console.log('Final list: ', LinkedList)
+    })
+  })
+
+  context('remove()', () => {
+    const myLinkedList = new LinkedList()
+    myLinkedList.insert('first')
+    myLinkedList.insert('second')
+
+    it('Removes the tail node from the list', () => {
+      expect(() => myLinkedList._remove())
+        .to.alter(() => myLinkedList._size(), {from: 2, to: 1})
+      expect(myLinkedList.list.data)
+        .to.eql('first')
+    })
+    it('Removes the value if there is only one value in the list', () => {
+      expect(() => myLinkedList._remove())
+        .to.alter(() => myLinkedList._size(), {from: 1, to: 0})
+        expect(myLinkedList.list.data)
+          .to.be.null
+    })
+  })
+
+  context('removeFirst()', () => {
+    const myLinkedList = new LinkedList()
+    myLinkedList.insert('first')
+    myLinkedList.insert('second')
+
+    it('Removes the head node from the list', () => {
+      expect(() => myLinkedList._removeFirst())
+        .to.alter(() => myLinkedList._size(), {from: 2, to: 1})
+      expect(myLinkedList.list.data)
+        .to.eql('second')
+    })
+    it('Removes the value if there is only one value in the list', () => {
+      expect(() => myLinkedList._removeFirst())
+        .to.alter(() => myLinkedList._size(), {from: 1, to: 0})
+      expect(myLinkedList.list.data)
+        .to.be.null
+    })
+})
+
+  context('isEmpty()', () => {
+    const myLinkedList = new LinkedList()
+
+    it('', () => {
+    })
+  })
+
+  context('size()', () => {
+    const myLinkedList = new LinkedList()
+    myLinkedList.insert('foo')
```

Comment:
Again, using different naming conventions means I have to 'reset' my brain to your new thought process.